controlplane/kubeadm/internal/filters.go

```diff
@@ -196,6 +196,15 @@ func cleanupConfigFields(kcpConfig *bootstrapv1.KubeadmConfigSpec, machineConfig
 	kcpConfig.ClusterConfiguration = nil
 	machineConfig.Spec.ClusterConfiguration = nil
 
+	// We have to treat the "empty" and CloudConfig format the same
+	// otherwise defaulting on KCP will trigger a machine rollout.
```

Comment:
```suggestion
	// We have to treat the "empty" and default value of format (CloudConfig) the same
	// otherwise defaulting on KCP will trigger a machine rollout.
```

nit: reword to make sure that empty and _default value_ of format are treated equal.